### PR TITLE
feat: multiple reference lines

### PR DESCRIPTION
**Pull request by rephus**
Closes: #3959

- [x]  remove the tick box and just keep the add option 
- [x] +Add like in the layout
-  adding/removing groups causes the line to disappear : created a separate issue for this 
- [x] update the red garbage can to this x icon so it matches conditional table formatting

### Description:

Screenshot from 2023-01-10 08-40-24
[Screencast from 10-01-23 09:07:26.webm](https://user-images.githubusercontent.com/1983672/211497072-cdd90b15-4f81-4598-9f23-401a67c9709d.webm)

[Screencast from 10-01-23 09:07:26.webm](https://user-images.githubusercontent.com/1983672/211497140-2a778c80-3023-4e4c-999d-03b5b12fdd77.webm)

### CODE REVIEW

**Review by TuringLovesDeathMetal — CHANGES_REQUESTED** [review]
This is so great @rephus! And so quickly built, well done 💪 

Just a few things I think could be nice to change (nothing major though):

1. Can we change the tick-box to be just `add reference line`? So, remove the tick box and just keep the `add` option (but update teh copy to `add reference line`:
image

The reason is: I just added like 4 reference lines, then I accidentally clicked the `include reference line` box, and _**all**_ of my reference lines just got deleted 😢 . I think it made more sense as an action when the only option was to add one line!

If we can make it look the same as the `add` option in the `layout` tab, that would be ideal 😚 👌 

2. adding/removing groups causes the line to disappear: https://www.loom.com/share/6d15b7583e324abea9985f69d140330a?from_recorder=1&focus_title=1

3. (I think you already saw this message) but update the red garbage can to this `x` icon so it matches conditional table formatting: 
image

**Comment by ZeRego** [comment]
@TuringLovesDeathMetal @rephus From a design perspective I would have the `arrow` icon to expand at the left side ( similar to how we have in the explore sidebar and the cards ) and that opens space for the delete button on the right side.

**Comment by rephus** [comment]
> @TuringLovesDeathMetal @rephus From a design perspective I would have the `arrow` icon to expand at the left side ( similar to how we have in the explore sidebar and the cards ) and that opens space for the delete button on the right side.

:+1:  Good suggestion

image

File: packages/common/src/types/savedCharts.ts [hunk 1]
```diff
@@ -75,17 +95,7 @@ export type Series = {
...
-        lineStyle: {
-            color: string;
-            width: number;
-            type: string;
-        };
-        label: {
-            formatter?: string;
-        };
-    };
+    markLine?: MarkLine;
```
Comment (ZeRego) [inline packages/common/src/types/savedCharts.ts:98]: What is the backwards compatibility of this change?

Comment (rephus) [reply packages/common/src/types/savedCharts.ts:98]: good question. it should be backwards compatible with the existing reference lines. I structured the types a bit better, and I added color and formatter to each markLineData.

Comment (rephus) [reply packages/common/src/types/savedCharts.ts:98]: I've just tested this. I created a chart with reference line and saved on master. 
Screenshot from 2023-01-10 16-13-03

Then I switched to my branch, and the chart was loaded with the same reference line. 

Screenshot from 2023-01-10 16-13-33

Comment (rephus) [reply packages/common/src/types/savedCharts.ts:98]: Fixed 

image

**Comment by IrakliJani** [comment]
@rephus nice work! 

few styling improvements that I have in mind:

1. you can use `FormGroup` and have labels on top rather than on the side. it will make the UI more consistent and input boxes aligned on the same line

CleanShot 2023-01-10 at 19 35 00@2x
vs
CleanShot 2023-01-10 at 19 35 18@2x

2. X button and collapse button are too big and and misaligned. you can use small prop for them and minimal styling for both, and add tooltips on what they do (for X maybe: "remove reference line").
CleanShot 2023-01-10 at 19 36 54@2x
CleanShot 2023-01-10 at 19 36 46@2x

3. one thing I am not sure of is the background for a group and I'm open for a discussion. For the context, I am building a similar inputs for conditional formatting and used a light background colour to represent a rule group

CleanShot 2023-01-10 at 19 39 17@2x

**Comment by PriPatel** [comment]
@IrakliJani yeah great point
The redesign will ensure better separation and distinction between groups **but** if its not too much work, I agree that a grey background for each group (like in Irakli's example) makes things a bit easier to follow.

**Comment by rephus** [comment]
Nice suggestions @IrakliJani , what do you think ? 

image

**Comment by IrakliJani** [comment]
> Nice suggestions @IrakliJani , what do you think ? image

for me X button looks too aggressive, I think minimal appearance would be better

**Review by ZeRego — APPROVED** [review]
code looks good 👍 

Minor style changes:
- grey box showing when rule is collapsed
- if there are multiple rules, show them collapsed by default

Screenshot 2023-01-11 at 14 44 26

**Pull request merged**